server/events/vcs/gitlab_client.go

```diff
@@ -212,7 +212,7 @@ func (g *GitlabClient) PullIsMergeable(repo models.Repo, pull models.PullRequest
 	}
 
 	for _, status := range statuses {
-		if !strings.HasSuffix(status.Name, fmt.Sprintf("/%s", command.Apply.String())) {
+		if !strings.HasPrefix(status.Name, fmt.Sprintf("atlantis/%s", command.Apply.String())) {
 			if !status.AllowFailure && project.OnlyAllowMergeIfPipelineSucceeds && status.Status != "success" {
 				return false, nil
 			}
```

Comment:
I feel like this is clearer:
```suggestion
		if strings.HasPrefix(status.Name, fmt.Sprintf("atlantis/%s", command.Apply.String())) {
			continue;
		}
		if !status.AllowFailure && project.OnlyAllowMergeIfPipelineSucceeds && status.Status != "success" {
			return false, nil
		}
```
To communicate that statusses with the `atlantis/apply` prefix are not considered whatsoever.

Additionally, I think there needs to be added a section here, alike this:
```
		if status.Status == "skipped" {
			continue;
		}
```
As currently `skipped` pipelines will also block for applying.